alerters/alerters-plugins/alerters-gnocchi/src/main/java/org/hawkular/alerter/gnocchi/GnocchiQuery.java

```diff
@@ -0,0 +1,403 @@
+        }
+        if (isEmpty(password)) {
+            password = this.properties.getOrDefault(GnocchiAlerter.PASSWORD, GnocchiAlerter.GNOCCHI_PASSWORD_DEFAULT);
+        }
+        basicAuth = new StringBuilder("Basic ")
+                .append(new String(Base64.getEncoder().encode(new StringBuilder()
+                        .append(user)
+                        .append(":")
+                        .append(password)
+                        .toString()
+                        .getBytes())))
+                .toString();
+        searchMetrics();
+        processMetricAggregation();
+        buildGnocchiQueries();
+    }
+
+    public void searchMetrics() {
+        // Case 1, fixed list of metric ids, so, no need to query names or anything
+        if (!isEmpty(metricIds)) {
+            String[] gnocchiIds = metricIds.split(",");
+            for (int i = 0; i < gnocchiIds.length; i++) {
+                nameIdMetrics.put(gnocchiIds[i].trim(), gnocchiIds[i].trim());
+            }
+            log.debugf("Gnocchi Metrics configured %s", nameIdMetrics);
+            return;
+        }
+
+        // Case 2, fixed list of metric names, a query to get gnocchi ids is necessary
+        if (!isEmpty(metricNames) && isEmpty(metricResourceQuery)) {
+            String[] splitMetricNames = metricNames.split(",");
+            List<String> gnocchiNames = new ArrayList<>();
+            for (int i = 0; i < splitMetricNames.length; i++) {
+                gnocchiNames.add(splitMetricNames[i].trim());
+            }
+            List rawAllMetrics = getAllMetrics();
+            for (int i = 0; i < rawAllMetrics.size(); i++) {
+                Map<String, String> metric = (Map<String, String>) rawAllMetrics.get(i);
+                if (gnocchiNames.contains(metric.get("name"))) {
+                    nameIdMetrics.put(metric.get("name"), metric.get("id"));
+                }
+            }
+            log.debugf("Gnocchi Metrics configured %s", nameIdMetrics);
+            return;
+        }
+
+        // Case 3, regexp defined for metrics names
+        if (!isEmpty(metricNamesRegexp) && isEmpty(metricResourceQuery)) {
+            Pattern pattern = Pattern.compile(metricNamesRegexp);
+            List rawAllMetrics = getAllMetrics();
+            for (int i = 0; i < rawAllMetrics.size(); i++) {
+                Map<String, String> metric = (Map<String, String>) rawAllMetrics.get(i);
+                if (pattern.matcher(metric.get("name")).matches()) {
+                    nameIdMetrics.put(metric.get("name"), metric.get("id"));
+                }
+            }
+            log.debugf("Gnocchi Metrics configured %s", nameIdMetrics);
+            return;
+        }
+
+        // Case 4, a resource query is defined, so metrics are filtered by this query
+        if (!isEmpty(metricResourceQuery)) {
+            List rawResourceMetrics = getResourceMetrics();
```

Comment:
That seems pretty wide as a query. You can use the search API to have a smaller set of results.

Reply:
+1 indeed it is used, read from the trigger configuration
https://github.com/hawkular/hawkular-alerts/pull/467/files/9348a574844de42dd4ef19ddaa853020c43b154a#diff-f7efcfbcd8455ebf114f6e51ced5131bR155
So, basically, it uses the search API to narrow the resources to fetch.